src/anemoi/training/diagnostics/callbacks/__init__.py

```diff
@@ -1081,58 +144,110 @@ def get_callbacks(config: DictConfig) -> list:  # noqa: C901
+                + f"This will be removed in a future release.\n Add the following to the `callbacks` list:\n{suggested_change}",
+                DeprecationWarning,
+            )
+            callbacks.append(callback_list(config))
 
-    if config.diagnostics.eval.enabled:
-        trainer_callbacks.append(RolloutEval(config))
+    for enable_key, callback_list in CONFIG_ENABLED_CALLBACKS:
+        if check_key(config, enable_key):
+            callbacks.append(callback_list(config))
 
-    if config.diagnostics.plot.enabled:
-        trainer_callbacks.extend(
-            [
-                PlotLoss(config),
-                PlotSample(config),
-            ],
-        )
-        if (config.diagnostics.plot.parameters_histogram or config.diagnostics.plot.parameters_spectrum) is not None:
-            trainer_callbacks.extend([PlotAdditionalMetrics(config)])
-        if config.diagnostics.plot.get("longrollout") and config.diagnostics.plot.longrollout.enabled:
-            trainer_callbacks.extend([LongRolloutPlots(config)])
+    return callbacks
 
-    if config.training.swa.enabled:
-        from pytorch_lightning.callbacks.stochastic_weight_avg import StochasticWeightAveraging
 
-        trainer_callbacks.append(
-            StochasticWeightAveraging(
-                swa_lrs=config.training.swa.lr,
-                swa_epoch_start=min(
-                    int(0.75 * config.training.max_epochs),
-                    config.training.max_epochs - 1,
-                ),
-                annealing_epochs=max(int(0.25 * config.training.max_epochs), 1),
-                annealing_strategy="cos",
-                device=None,
-            ),
-        )
+def get_callbacks(config: DictConfig) -> list[Callback]:  # noqa: C901
+    """Setup callbacks for PyTorch Lightning trainer.
+
+    Set `config.diagnostics.callbacks` to a list of callback configurations
+    in hydra form.
+
+    E.g.:
+    ```
+    callbacks:
+        - _target_: anemoi.training.diagnostics.callbacks.RolloutEval
+          rollout: 1
+          frequency: 12
+    ```
+
+    Set `config.diagnostics.plot.callbacks` to a list of plot callback configurations
+    will only be added if `config.diagnostics.plot.enabled` is set to True.
+
+    A callback must take a `DictConfig` in its `__init__` method as the first argument,
+    which will be the complete configuration object.
+
+    Some callbacks are added by default, depending on the configuration.
+    See CONFIG_ENABLED_CALLBACKS for more information.
+
+    Parameters
+    ----------
+    config : DictConfig
+        Job configuration
 
+    Returns
+    -------
+    List[Callback]
+        A list of PyTorch Lightning callbacks
+
+    """
+
+    trainer_callbacks: list[Callback] = []
+
+    # Get Checkpoint callback
+    checkpoint_callback = _get_checkpoint_callback(config)
+    if checkpoint_callback is not None:
+        trainer_callbacks.extend(checkpoint_callback)
+
+    # Base callbacks
```

Comment:
Apologies since I might just be slow, but based on this we are mixing different ways of instantiating the callbacks in the config right? Cause we have plotting callbacks and rollout where we use the target method, but then that's not the case for checkpoint and the ones about learning rate? My concern here is when extending to new callbacks it might difficult to decide which way to go, and probably not very intuitive for the user. Could we consider using the instantiation method for all callbacks to keep it consistent?

Reply:
I think it definitetly requires some discussion, 
- Some callbacks must be included based on configuration options, see swa, or UUID
- Others make sense to include by default, with a flag
- Others again, are up to the user to include.

The lines to be drawn are tricky, but I would suggest only a very core callback is added via the config enabled options, and the rest the _target_ approach

Reply:
I am leaning to agree that all but the essential should be done throught the _target_ method, including plots